src/SystemManager.cc

```diff
@@ -38,6 +53,30 @@ void SystemManager::LoadPlugin(const Entity _entity,
                                const std::string &_name,
                                const sdf::ElementPtr &_sdf)
 {
+  for (const auto & pluginInfo : loadedPlugins)
+  {
+    if (_entity == pluginInfo.entity &&
+        _name == pluginInfo.name)
+    {
+      if (_name == "gz::sim::systems::UserCommands" ||
+          _name == "gz::sim::systems::SceneBroadcaster" ||
+          _name == "gz::sim::systems::Physics")
+      {
+        //  Plugin already loaded
+        igndbg << "This system Plugin [" << _name
+               << "] is already loaded in this entity [" << _entity
+               << "]" << std::endl;
+        return;
+      }
+    }
+  }
```

Comment:
Can you put a comment on what this loop is doing?  It's not particularly clear to me.

Also, the UserCommands/SceneBroadcaster/Physics systems are the ones that we have by default in the config file, but users are able to define their own set of default plugins in their configuration files.  Do you know what the interaction is in that case?

Reply:
I think I added this because of the issue in the reset, I will revisit this again